web/app/Models/User.php

```diff
@@ -120,34 +123,33 @@ public function followingUsers() : Collection
         return $this->my(InteractionType::USER_FOLLOWS_USER);
     }
 
-    /**
-     * Retrieve the users following this user.
-     *
-     * @return Collection<User>
-     */
-    public function followers() : Collection
-    {
-        return $this->their(InteractionType::USER_FOLLOWS_USER);
-    }
+    /**************************************************
+     * Following Users
+     *************************************************/
+
 
     /**
      * Follow a user.
      * @param User $user_to_follow
-     * @return Interaction
+     * @return bool
      */
-    public function followUser(User $user_to_follow) : Interaction
+    public function followUser(User $user_to_follow) : bool
     {
-        return Interaction::add($this, InteractionType::USER_FOLLOWS_USER, $user_to_follow);
+        if ($this->isFollowingUser($user_to_follow) === false)
+        {
+            return Interaction::create($this, InteractionType::USER_FOLLOWS_USER, $user_to_follow);
```

Comment:
For this an several other interaction methods: Should these be in transactions to avoid data races?

Reply:
Yes. I had it in my head that I couldn't mix transactions and Eloquent but that doesn't look to be the case, I will add these where it makes sense to before I merge.